tests/e2e/buy.spec.ts

```diff
@@ -0,0 +1,33 @@
+import { test, expect } from '@playwright/test';
+import { FlightBookingPage } from 'e2e-tests/pom/buy.page';
+import { LoginPage } from 'e2e-tests/pom/login.page';
+import { LOGIN_DETAILS, FLIGHT_DETAILS, PASSENGER_DETAILS, CARD_DETAILS } from '../consts';
+
+let loginPage: LoginPage;
+let flightBookingPage: FlightBookingPage;
+
+test.describe('Travel Agileway E2E Tests', () => {
+  test.beforeEach('setup', async ({page}) => {  
+      loginPage = new LoginPage(page);
+      flightBookingPage = new FlightBookingPage(page);
+
+      await flightBookingPage.goTo();
+      await loginPage.login(LOGIN_DETAILS);
+      await expect(page).toHaveURL('https://travel.agileway.net/flights/start');
+      await expect(page.locator('#flash_notice')).toContainText('Signed in!');
+      await expect(page.locator('h2')).toContainText('Select Flight');
+  })
+
+  test('Purchase a reservation', async ({ page }) => {
+    const flightBookingPage = new FlightBookingPage(page);
```

Comment:
Since flightBookingPage is already declared in the beforeEach, there's no need to declare it again here. Just use the existing one.